src/brogue/Globals.c

```diff
@@ -589,8 +589,8 @@ const floorTileType tileCatalog[NUMBER_TILETYPES] = {
     {G_GRASS,    &foliageColor,   0,                      60, 15, DF_PLAIN_FIRE,  0,          DF_FOLIAGE_REGROW, 100,         NO_LIGHT,       (T_IS_FLAMMABLE), (TM_VANISHES_UPON_PROMOTION),                                                     "trampled foliage",     "dense foliage fills the area, thriving on what sunlight trickles in."},
     {G_FOLIAGE,  &fungusForestLightColor,0,                      45, 15, DF_PLAIN_FIRE,  0,          DF_TRAMPLED_FUNGUS_FOREST, 0,   FUNGUS_FOREST_LIGHT,(T_OBSTRUCTS_VISION | T_IS_FLAMMABLE), (TM_STAND_IN_TILE | TM_VANISHES_UPON_PROMOTION | TM_PROMOTES_ON_STEP),"a luminescent fungal forest", "luminescent fungal growth fills the area, groping upward from the rich soil."},
     {G_GRASS,    &fungusForestLightColor,0,               60, 15, DF_PLAIN_FIRE,  0,          DF_FUNGUS_FOREST_REGROW, 100,   FUNGUS_LIGHT,   (T_IS_FLAMMABLE), (TM_VANISHES_UPON_PROMOTION),                                                     "trampled fungal foliage", "luminescent fungal growth fills the area, groping upward from the rich soil."},
-    {G_CRYSTAL,  &forceFieldColor,       &forceFieldColor,       0,  0,  0,              0,          DF_FORCEFIELD_MELT, -200,       FORCEFIELD_LIGHT, (T_OBSTRUCTS_PASSABILITY | T_OBSTRUCTS_GAS | T_OBSTRUCTS_DIAGONAL_MOVEMENT), (TM_STAND_IN_TILE | TM_VANISHES_UPON_PROMOTION | TM_PROMOTES_ON_STEP),       "a green crystal",      "The translucent green crystal is melting away in front of your eyes."},
```

Comment:
Do you know why `TM_PROMOTES_ON_STEP` was already here for one of these lines?

Reply:
Wait, I bet this is shattering crystal, which already did not encase things.  Is that it?

Reply:
> Do you know why `TM_PROMOTES_ON_STEP` was already here for one of these lines?

There are actually two crystal states: regular obstruction crystal, and dissolving obstruction crystal (same background appearance but black '#' character). Dissolving obstruction crystal lasts exactly one turn when it is exposed to air. It gives some warning when a crystal tile is about to dissolve. Putting TM_PROMOTES_ON_STEP in regular obstruction crystal ensured that crystals containing monsters (or, unintentionally, items) would immediately revert to dissolving crystal, so that creatures in crystal always break out one turn after being exposed to open air. It also meant that the location of obstructed phantoms was apparent (just look for the interior crystal cells with black forecolor characters) but I don't think anyone ever picked up on that.

The new scheme has creatures instantly break out whether or not they're exposed to air so all of this is obviated.

Reply:
> location of obstructed phantoms was apparent (just look for the interior crystal cells with black forecolor characters) but > I don't think anyone ever picked up on that.

I made ample use of this in my weekend contest run. I could easily deal with otherwise problematic phantoms because the black forecolor exposes their location and when adjacent, you can deal staff damage. :)

Reply:
So is one of the effects of this change that crystals no longer dissolve when items are in them?

Reply:
Correct, they don't preferentially dissolve around items.